src/main/java/uk/gov/ons/ctp/common/rest/RestClient.java

```diff
@@ -144,19 +144,49 @@ public RestTemplate getRestTemplate() {
       MultiValueMap<String, String> queryParams,
       Object... pathParams)
       throws ResponseStatusException {
-    log.debug("Enter getResources for path: {}", path);
+    return doHttpOperation(
+        HttpMethod.GET, path, null, clazz, headerParams, queryParams, pathParams);
+  }
+
+  /**
+   * Use to perform an http operation GET/PUT/POST to retrieve a single resource
+   *
+   * @param <T> the type that will returned by the server we call
+   * @param <O> is the type of payload to send.
```

Comment:
Since O is often confused with Zero, can I suggest using <S> instead? Same comment thoughout this file.

Reply:
Never personally mistaken it, but changed to <P> for payload.